packages/app-builder-lib/src/targets/snap.ts

```diff
@@ -214,6 +213,16 @@ export default class SnapTarget extends Target {
       return
     }
 
+    if (this.isUseTemplateApp) {
+      // remove fields that are valid in snapcraft.yaml, but not snap.yaml
+      delete snap.compression;
+      delete snap.contact;
+      delete snap.donation;
+      delete snap.issues;
+      delete snap['source-code'];
+      delete snap.website;
+    }
```

Comment:
Just so I understand this change further, what's the difference in usage between snapcraft.yaml and snap.yaml?

Reply:
Also, to be more clear about intent, thoughts on adjusting the logic to:
```
// remove fields that are valid in snapcraft.yaml, but not snap.yaml
const forbiddenSnapFields = ['compression', 'donation', 'issues', 'source-code', 'website']
for (field of forbiddenSnapFields) {
   delete snap[field]
}
```
Could be `of` or `in`, I always forget the difference when reviewing PRs on-the-fly

Reply:
AFAIK the `meta/snap.yaml` file is present in the built apps directory before it gets run through squashfs, and so is present in the final snap package when it is released; `snapcraft.yaml` seems to be used to tell the snap cli tool how to build the snap. The two files are very similar, but don't have exactly the same schema. It looks like `electron-builder` was removing the `parts` field, but not some of the newer fields that were added more recently to snap.

Reply:
Additional note, we have this at Line 178
```
if (this.isUseTemplateApp) {
      delete snap.parts
}
```
Should we incorporate your additional logic there or is it essential it be placed down here?

Reply:
For reference, here are the documented schemas:
- https://snapcraft.io/docs/snap-format
- https://snapcraft.io/docs/snapcraft-yaml-reference

Reply:
> Additional note, we have this at Line 178
> 
> ```
> if (this.isUseTemplateApp) {
>       delete snap.parts
> }
> ```
> 
> Should we incorporate your additional logic there or is it essential it be placed down here?

I have stripped right before we write the file: this is necessary for example with the `compression` field, as `--compression` is only pushed to `args` if `snap.compression` is truthy, so deleting the field early would cause incorrect behavior. I could move the delete of parts down into the if block I just added though.